central/cve/matcher/matcher.go

```diff
@@ -127,15 +127,7 @@ func (m *CVEMatcher) IsClusterAffectedByK8sCVE(_ context.Context, cluster *stora
 
 // IsClusterAffectedByIstioCVE returns true if cluster is affected by istio cve
 func (m *CVEMatcher) IsClusterAffectedByIstioCVE(ctx context.Context, cluster *storage.Cluster, cve *schema.NVDCVEFeedJSON10DefCVEItem) (bool, error) {
-	ok, err := m.isIstioControlPlaneRunning(ctx)
-	if err != nil {
-		return false, err
-	}
-	if !ok {
-		return false, nil
-	}
-
-	versions, err := m.getAllIstioComponentsVersionsInCluster(ctx, cluster)
+	versions, err := m.GetValidIstioVersions(ctx, cluster)
```

Comment:
after checking for an error, maybe we should ensure we don't bother continuing if `versions` is empty